metaflow/datastore/s3_backend.py

```diff
@@ -101,39 +107,39 @@ def list_content(self, paths):
             return [self.list_content_result(
                 path=o.url[strip_prefix_len:], is_file=o.exists) for o in results]
 
-    def save_bytes(self, path_and_bytes, overwrite=False):
+    def save_bytes(self, path_and_bytes_iter, overwrite=False, len_hint=0):
         """
         Creates objects and stores them in the datastore.
 
         If overwrite is False, any existing object will not be overwritten and
         will be silently ignored
 
-        The objects are specified in the objects dictionary where the key is the
-        path to store the object and the value is a file-like object from which
-        bytes can be read.
+        The objects are specified in an iterator over (key, path) tuples where
```

Comment:
nit: this is confusing: "key is the path to store" and "the value (ie: the name 'path') is a file-like object". I would rename the iterator to be over (path, file-like-object) or (path, fileobj) or something like that.